genet/core.py

```diff
@@ -2263,3 +2263,58 @@ def get_link_slope_dictionary(self, elevation_dict):
             slope_dict[link_id] = {'slope': link_slope}
 
         return slope_dict
+
+    def summary(self):
+        """
+        Returns a report with summary statistics for the network and the schedule.
+        """
+        logging.info('Creating a summary report')
+        report = {}
+
+        network_stats = {'Number of network links': nx.number_of_nodes(self.graph),
+                         'Number of network nodes': nx.number_of_edges(self.graph)}
+        report['network_graph_info'] = network_stats
+        report['modes'] = {}
+        report['modes']['Modes on network links'] = self.modes()
+
+        highway_tags = self.link_attribute_data_under_key({'attributes': 'osm:way:highway'})
```

Comment:
need to add something for 'backwards' compatibility here. It is still possible to read additional attributes in long form, in which case the query would be: `{'attributes': {'osm:way:highway': 'text'}}`. I think there is a simple way to do this, just by checking if `highway_tags` is empty, add a warning and attempt to search using long form --- but you will need to do a test to see if that's right

Reply:
Yep that's a really good point, I have added a check for that so that both cases can be handled